File: elixir/7ku/mumbling.ex
defmodule Mumbling do
  def accum(s) do
    Enum.map(
      1..String.length(s),
      &(String.duplicate(String.at(s, &1 - 1), &1) |> String.capitalize())
    )
    |> Enum.join("-")
  end
end

File: ruby/7ku/Mumbling.rb
# frozen_string_literal: true

# Mumbling
# https://www.codewars.com/kata/5667e8f4e3f572a8f2000039

def accum(s)
  s.chars.map.with_index { |char, index| (char.to_s * (index + 1)).downcase.capitalize }.join('-')
end

p accum('HbideVbxncC') # , "H-Bb-Iii-Dddd-Eeeee-Vvvvvv-Bbbbbbb-Xxxxxxxx-Nnnnnnnnn-Cccccccccc-Ccccccccccc"
